pallets/staking/src/lib.rs

```diff
@@ -2147,22 +2147,24 @@ impl<T: Trait> Module<T> {
             }
         }
 
-        let nominator_votes = <Nominators<T>>::enumerate().map(|(nominator, nominations)| {
-            let Nominations {
-                submitted_in,
-                mut targets,
-                suppressed: _,
-            } = nominations;
-
-            // Filter out nomination targets which were nominated before the most recent
-            // non-zero slash.
-            targets.retain(|stash| {
-                <Self as Store>::SlashingSpans::get(&stash)
-                    .map_or(true, |spans| submitted_in >= spans.last_nonzero_slash())
-            });
+        let nominator_votes = <Nominators<T>>::enumerate()
+            .filter(|(nominator, _)| Self::is_validator_or_nominator_compliant(&nominator))
```

Comment:
In addition to filtering, we also need to remove the nominators from the system. Otherwise, they will still earn rewards (and get slashed).

Reply:
They will never earn rewards or neither slashed as they will never be a part of `IndividualExposure` for the validators/operators. It will be just like extra data in the storage. I avoid deleting here because that makes a process bit heavier, I am not worried about the redundant data for v1 as process gonna change post v1.